File: app/src/main/java/com/weberbox/pifire/service/NotificationServiceExtension.java
package com.weberbox.pifire.service;

import android.app.PendingIntent;
import android.content.Context;
import android.content.Intent;

import androidx.core.app.NotificationCompat;
import androidx.core.content.ContextCompat;

import com.onesignal.OSMutableNotification;
import com.onesignal.OSNotification;
import com.onesignal.OSNotificationReceivedEvent;
import com.onesignal.OneSignal.OSRemoteNotificationReceivedHandler;
import com.weberbox.pifire.MainActivity;
import com.weberbox.pifire.R;

@SuppressWarnings("unused")
public class NotificationServiceExtension implements OSRemoteNotificationReceivedHandler {

    @Override
    public void remoteNotificationReceived(Context context, OSNotificationReceivedEvent
            receivedEvent) {
        OSNotification notification = receivedEvent.getNotification();
        Intent intent = new Intent(context, MainActivity.class);
        intent.addFlags(Intent.FLAG_ACTIVITY_SINGLE_TOP);

        PendingIntent pendingIntent = PendingIntent.getActivity(context, 0, intent,
                PendingIntent.FLAG_ONE_SHOT | PendingIntent.FLAG_IMMUTABLE);

        OSMutableNotification mutableNotification = notification.mutableCopy();
        mutableNotification.setExtender(builder -> {
            builder.setColor(ContextCompat.getColor(context, R.color.colorAccent));
            builder.setSmallIcon(R.drawable.ic_grill);
            builder.setAutoCancel(true);
            builder.setVisibility(NotificationCompat.VISIBILITY_PUBLIC);
            builder.setOnlyAlertOnce(false);
            builder.setCategory(NotificationCompat.CATEGORY_MESSAGE);
            builder.setWhen(System.currentTimeMillis());
            builder.setContentTitle(notification.getTitle());
            builder.setContentText(notification.getBody());
            builder.setContentIntent(pendingIntent);
            return builder;
        });

        receivedEvent.complete(mutableNotification);
    }
}

File: app/src/main/java/com/weberbox/pifire/core/service/NotificationServiceExtension.kt
package com.weberbox.pifire.core.service

import android.app.PendingIntent
import android.app.TaskStackBuilder
import android.content.Context
import android.content.Intent
import androidx.core.app.NotificationCompat
import com.onesignal.OSNotificationReceivedEvent
import com.onesignal.OneSignal.OSRemoteNotificationReceivedHandler
import com.weberbox.pifire.MainActivity
import com.weberbox.pifire.R

@Suppress("unused")
class NotificationServiceExtension : OSRemoteNotificationReceivedHandler {

    override fun remoteNotificationReceived(
        context: Context,
        receivedEvent: OSNotificationReceivedEvent
    ) {
        val notification = receivedEvent.notification

        val activityIntent = Intent(context, MainActivity::class.java).apply {
            action = Intent.ACTION_VIEW
        }

        val pendingIntent = TaskStackBuilder.create(context).run {
            addNextIntentWithParentStack(activityIntent)
            getPendingIntent(0, PendingIntent.FLAG_UPDATE_CURRENT or PendingIntent.FLAG_IMMUTABLE)
        }

        val mutableNotification = notification.mutableCopy()
        mutableNotification.setExtender { builder: NotificationCompat.Builder ->
            builder.setSmallIcon(R.drawable.ic_grill)
            builder.setAutoCancel(true)
            builder.setVisibility(NotificationCompat.VISIBILITY_PUBLIC)
            builder.setOnlyAlertOnce(false)
            builder.setCategory(NotificationCompat.CATEGORY_NAVIGATION)
            builder.setWhen(System.currentTimeMillis())
            builder.setContentTitle(notification.title)
            builder.setContentText(notification.body)
            builder.setContentIntent(pendingIntent)
        }

        receivedEvent.complete(mutableNotification)
    }
}